container/shim/src/index.js

```diff
@@ -119,27 +134,36 @@ const handleCID = asyncHandler(async (req, res) => {
     return
   }
 
+  respondFromIPFSGateway(req, res, { cid, format })
+})
+
+// Whenever nginx doesn't have a CAR file in cache, this is called
+app.get('/ipfs/:cid', handleCID)
+app.get('/ipfs/:cid/:path*', handleCID)
+
+function respondFromIPFSGateway (req, res, { cid, format }) {
   debug(`Fetch ${req.path} from IPFS`)
 
-  const ipfsUrl = new URL(IPFS_GATEWAY_ORIGIN + req.path)
+  const ipfsUrl = new URL(IPFS_GATEWAY_ORIGIN + toUtf8(req.path))
   if (format) {
     ipfsUrl.searchParams.set('format', format)
   }
-  for (const key of ['filename', 'download']) {
-    if (key in req.query) {
-      ipfsUrl.searchParams.set(key, req.query[key])
-    }
+  for (const [key, val] of Object.entries(req.query)) {
```

Comment:
should be safe to pass on I think?

Reply:
what do you mean?

Reply:
Before this change, only some query parameters were passed on, now all are passed on. I was wondering if it is safe to do this 🤔

Reply:
hmm yeah, I think it's fine. What's the worse that could happen? (famous last words)

Reply:
Agreed! :D